src/mpc/faand.rs

```diff
@@ -1080,15 +1315,24 @@ pub(crate) async fn beaver_aand(
     Ok(alpha_and_beta)
 }
 
-/// Check and return d-values for a vector of shares.
+/// Computes and verifies the d-values used for bucket combination in Pi_aAND.
+///
+/// For each bucket of leaky AND triples, this function:
+/// 1. Computes local `d`-values as XORs of the `y`-shares within the bucket.
+/// 2. Collects corresponding MACs from all parties.
+/// 3. Verifies that the MACs are consistent with the global MAC key delta.
+/// 4. Aggregates all parties' contributions to obtain the final d-values.
+///
+/// These d-values are used to safely combine leaky AND triples into a single
+/// non-leaky authenticated AND.
 #[instrument(level=Level::DEBUG, skip_all, err)]
 async fn check_dvalue(
     (channel, delta): (&impl Channel, Delta),
     i: usize,
     n: usize,
     buckets: &[Bucket<'_>],
 ) -> Result<Vec<Vec<bool>>, Error> {
-    // Step (a) compute and check macs of d-values.
+    // Step (a) Compute and verify d-values for each bucket.
```

Comment:
In this function, several times the first element of each bucket is XORed with each following element, resulting in vectors of size `b-1`. Can't we aggregate each bucket into one `d` share which we reveal and check the MAC of? If secure, this would significantly reduce the communication from this method and likely also reduce the computation.

Reply:
This is wrong, see the comment in `combine_bucket`.